src/array.rs

```diff
@@ -189,8 +190,10 @@ impl<T, D> IntoPy<PyObject> for PyArray<T, D> {
 }
 
 impl<'py, T: Element, D: Dimension> FromPyObject<'py> for &'py PyArray<T, D> {
-    fn extract(ob: &'py PyAny) -> PyResult<Self> {
+    fn extract_bound(ob: &Bound<'py, PyAny>) -> PyResult<Self> {
         PyArray::extract(ob)
+            .map(Clone::clone)
```

Comment:
```suggestion
            .cloned()
```

Reply:
I tried that as well, but CI was not happy about it 🤣

Reply:
Indeed. But when I tried to fix up the formatting locally, nightly Clippy presented me with

```console
error: current MSRV (Minimum Supported Rust Version) is `1.56.0` but this item is stable since `1.59.0`
   --> src/array.rs:195:30
    |
195 |         PyArray::extract(ob).cloned().map(Bound::into_gil_ref)
    |                              ^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#incompatible_msrv
    = note: `-D clippy::incompatible-msrv` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::incompatible_msrv)]`
```

which would be a real life saver when maintaining crates with a rather old MSRV like these here.

Reply:
That would be really handy to have on stable.